hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala

```diff
@@ -15,23 +15,22 @@ import scala.language.{existentials, postfixOps}
 
 class UnsupportedExtraction(msg: String) extends Exception(msg)
 
-case class Aggs(postAggIR: IR, init: IR, seqPerElt: IR, aggs: Array[PhysicalAggSignature]) {
-  val typ: PTuple = PTuple(aggs.map(Extract.getPType): _*)
+case class Aggs(postAggIR: IR, init: IR, seqPerElt: IR, aggs: Array[AggStateSignature]) {
   val nAggs: Int = aggs.length
 
   def isCommutative: Boolean = {
-    def aggCommutes(agg: PhysicalAggSignature): Boolean = agg.nested.forall(_.forall(aggCommutes)) && AggIsCommutative(agg.op)
+    def aggCommutes(agg: AggStateSignature): Boolean = AggIsCommutative(agg.default) && agg.nested.forall(_.forall(aggCommutes))
```

Comment:
I don't fully understand why this couldn't lead to problems. If my default is ArrayElementLengthCheck and my other operation is a non-commutative operation (maybe something involving an ordering step), will this work?

Reply:
You're right, I'll change this to check `agg.m.forall` instead of just the default. Would work now, but you are certainly correct to future-proof it.